app-text/mandoc/mandoc-1.14.5-r1.ebuild

```diff
@@ -0,0 +1,100 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs
+
+DESCRIPTION="Suite of tools compiling mdoc and man"
+HOMEPAGE="https://mdocml.bsd.lv/"
+SRC_URI="https://mdocml.bsd.lv/snapshots/${P}.tar.gz"
+
+LICENSE="ISC"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+IUSE="system-man"
+
+RDEPEND="sys-libs/zlib
+	system-man? ( !sys-apps/man-db )
+"
+DEPEND="${RDEPEND}"
+
+pkg_setup() {
+	if use system-man ; then
+		# allow uncompressed and gzip
+		[[ ! -z "${PORTAGE_COMPRESS+unset}" ]] && \
+		( [[ "${PORTAGE_COMPRESS}" == "gzip" ]] || [[ "${PORTAGE_COMPRESS}" == "" ]] ) || \
+		die "only PORTAGE_COMPRESS=gzip or '' is allowed for system wide support"
```

Comment:
What about alternative package managers which do not use `PORTAGE_COMPRESS` variable?
Maybe just print a message without calling `die`?

Reply:
Not sure that die is appropriate in any case, maybe an error or ewarn is enough. If it's going to die then you could at least add a way out with `[[ -n ${I_KNOW_WHAT_I_AM_DOING} ]]`.

The check should go in pkg_pretend. Instead of a subshell you can do this:
```
[[ "${PORTAGE_COMPRESS}" == "gzip" || "${PORTAGE_COMPRESS}" == "" ]]
```

Reply:
good point!
done.